frontend/src/views/camp/Invitation.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <auth-container>
+    <div v-if="campCollaborationFound && campCollaborationOpen !== false">
+      <h1 class="display-1">{{ this.$tc('components.invitation.title') }} {{ camp ? camp.title : '' }}</h1>
+
+      <v-spacer />
+      <div v-if="isLoggedIn">
+        <v-btn color="primary"
+               x-large
+               class="my-4" block
+               @click="acceptInvitation">
+          {{ this.$tc('components.invitation.acceptCurrentAuth') }}
+        </v-btn>
+        <v-btn color="primary"
+               x-large
+               class="my-4" block
+               @click="useAnotherAccount">
+          {{ this.$tc('components.invitation.useOtherAuth') }}
+        </v-btn>
+      </div>
+      <div v-else>
+        <v-btn color="primary"
+               x-large
+               class="my-4" block
+               :to="loginLink">
+          {{ this.$tc('components.invitation.login') }}
+        </v-btn>
+        <v-btn color="primary"
+               x-large
+               class="my-4" block
+               :to="{ name: 'register' }">
+          {{ this.$tc('components.invitation.register') }}
+        </v-btn>
+      </div>
+      <v-btn color="red"
+             x-large
+             class="my-4" block>
+        {{ this.$tc('components.invitation.reject') }}
+      </v-btn>
+    </div>
+    <v-alert v-else type="error">
+      {{ this.$tc('components.invitation.notFound') }}
+    </v-alert>
+  </auth-container>
+</template>
+
+<script>
+import AuthContainer from '@/components/layout/AuthContainer'
+import { campRoute, loginRoute } from '@/router'
+
+function loginLink () {
+  return loginRoute(this.$route.fullPath)
+}
+
+export default {
+  name: 'Invitation',
+  components: { AuthContainer },
+  props: {
+    getCamp: { type: Function, required: true },
+    getCampCollaboration: { type: Function, required: true }
+  },
+  data () {
+    return {
+      camp: undefined,
+      campCollaboration: undefined
+    }
+  },
+  computed: {
+    campCollaborationFound () {
+      return this.campCollaboration !== null
+    },
+    campCollaborationOpen () {
+      if (!this.campCollaboration) {
+        return undefined
+      }
+      return this.campCollaboration.status === 'invited'
+    },
+    isLoggedIn () {
+      return this.$auth.isLoggedIn()
+    },
+    loginLink
+  },
+  mounted () {
+    const me = this
```

Comment:
Ich wollte sowohl camp als auch campCollaboration als Promise hineingeben.
Vue wollte das aber nicht, deshalb habe ich das jetzt so unschön gelöst.
Hat jemand eine bessere Idee?

Ist es besser, blockierend auf die Requests zu warten und erst dann die Komponente zu rendern?

Reply:
Du kannst einerseits wenn das nötig ist im Router auf die CampCollaboration warten, so wie wir das schon z.B. mit `requirePeriod` machen. Das wäre die Variante "blockierend auf die Requests zu warten", aber da würde ich nur auf das absolut nötigste warten (also die Collaboration ja, aber das Camp noch nicht zwingend).

Als weitere Massnahme würde ich dann in `campCollaborationFromInviteKey` nicht eine Promise-Funktion zurückgeben, sondern eine Funktion die direkt die Entity aus dem Store zurückgibt. Der Store gibt auch etwas "brauchbares" zurück (ein LoadingProxy / LoadingStoreValue, quasi ein intelligenter Platzhalter) während die Entity noch am laden ist. Wenn du darauf unbedingt warten willst / musst würde ich das erst in Invitation.vue machen, nicht schon im Router. In Invitation.vue kannst du dann mit `this.campCollaboration._meta.loading` prüfen, ob das Objekt noch am laden ist.

Blöd ist halt dass `/api/camp-collaborations?inviteKey=xyz` eine Collection zurückgibt, und du dann noch das erste Element da rausfischen musst. Wäre es eine Möglichkeit, `inviteKey` in der Datenbank unique zu machen und bei diesem Query nur die eine passende `CampCollaboration` zurückzugeben (oder 404 im Fehlerfall)? Oder falls du das eine schlechte Idee findest, auf einem Collection Endpoint je nach Query Parameter nur eine einzelne Entity zurückzugeben, wie wäre es mit einem separaten Endpoint dafür? Etwa `/api/camp-collaboration-by-invite-key/{inviteKey}` oder `/api/invitations/{inviteKey}`?

Reply:
> Looking good! I have a few basic topics that I'd like to discuss before doing a very detailed code review.

Before i change the retrieval of the campCollaboration and the patch after the user accepts or rejects to RCP.
Do you want this in the current PR, or do we merge the current PR and I create a new one where i move the functionality to RCP?
I would rather merge the current PR (if it's ok).